### PR TITLE
DEVDOCS-4129: [net new] BODL overview for GA4

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4129]

Related [PR ](https://github.com/bigcommerce-labs/next-dev-center-prototype/pull/153)for redirects 

[DEVDOCS-4129]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/api-docs/analytics/bodl-ga4.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+> Stencil merchants can turn off the BODL and deploy their own custom version of the BODL. In this scenario, the merchant is choosing a custom route and is expected to ensure that analytics tracking scripts still work.
+
+## Analytic tracking scripts listen for events 
+
+You can pull storefront data from BODL to your solution using analytic storefront tracking scripts. Your analytic scripts will need to listen to the event and take in data that are stored in BODL.  
+
+Below is example code for your analytics script consuming the event. For more info, consult the MDN website doc [Add Event Listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener).
+
+```js title="Add event listener" lineNumbers
+document.addEventListener('checkout_begin', function (bodl_event) {
```
Comment (solofeed) [inline docs/api-docs/analytics/bodl-ga4.md:138]: @jairo-bc Should we mention we have two strategies here and recommended one?

- subscribe via `bodlEvents` - `window.bodlEvents.checkout.checkoutBegin`
- subscribe via `document` - `document.addEventListener('checkout_begin')`

and what the differences and advantages

File: docs/api-docs/analytics/bodl-ga4.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
+# Big Open Data Layer
+
+Big Open Data Layer (shortened as `BODL`, pronounced 'Bottle') is a global JavaScript object that allows BigCommerce to integrate with third-party analytics providers. This datalayer collects storefront data on shopper behavior and holds data needed for an analytic provider, such as GA4, to make analytic reports.  With BODL, providers can easily view and manage captured data from a standardized source. Providers can reformat the event data for their solutions.
+
+This guide demonstrates how you can integrate a BigCommerce store with Google Analytics (GA4). Using BODL, you can track when a shopper starts checkout and purchases an order. You can track events for both redirected and embedded checkout.
```
Comment (solofeed) [inline docs/api-docs/analytics/bodl-ga4.md:5]: > You can track events for both redirected and embedded checkout.

I think if we mention embedded checkout here, we should explain how it works

https://bigcommercecloud.atlassian.net/browse/KB-15839 here we started topic about it

**Review by solofeed — DISMISSED** [review]
LGTM! 🚀 
Only a few minor clarifications

File: docs/api-docs/analytics/bodl-ga4.md (hunk 1 above)
Comment (solofeed) [reply docs/api-docs/analytics/bodl-ga4.md:138]: Also, secure way to subscribe only after the full page loads after all scripts

```
if (window.addEventListener) {
    window.addEventListener("load", functionWithCustomMerchantSubscribers, false)
}
```

**Review by bc-rmalyavc — APPROVED** [review]
Looks good. LGTM

File: docs/api-docs/analytics/bodl-ga4.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
+# Big Open Data Layer
+
+Big Open Data Layer (shortened as `BODL`, pronounced 'Bottle') is a global JavaScript object that allows BigCommerce to integrate with third-party analytics providers. This datalayer collects storefront data on shopper behavior and holds data needed for an analytic provider, such as GA4, to make analytic reports.  With BODL, providers can easily view and manage captured data from a standardized source. Providers can reformat the event data for their solutions.
+
+This guide demonstrates how you can integrate a BigCommerce store with [Google Analytics (GA4)](https://support.bigcommerce.com/s/article/GA4). Using BODL, you can track when a shopper starts checkout and purchases an order. You can track events for both redirected and embedded checkout.
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/analytics/bodl-ga4.md:5]: ^URL to the KB doc will work once KB article is published

File: docs/api-docs/analytics/bodl-ga4.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+| `discount` | number | Discount applied to purchase. | [Get a checkout](/api-reference/store-management/checkouts/checkout/checkoutsbycheckoutidget) <br> `data.cart.discount_amount` | `discount` | N |
+| `index` | integer | Position of item in a list. | - | `index` | N |
+| `brand_name` | string | Brand name. | [Get a product](/api-reference/store-management/catalog/products/getproductbyid) <br> `data.brand_name` | `brand` | N |
+| `category_names[]` | array | Category names. | [Get a product](/api-reference/store-management/catalog/products/getproductbyid) <br> `data.categories[]` | `item_category` | N |
+
+
+## Resources
+- [Checkout SDK JS](https://github.com/bigcommerce/checkout-sdk-js)
+- [Checkout JS](https://github.com/bigcommerce/checkout-js)
+- [Google Analytics 4 support article](https://support.bigcommerce.com/s/article/GA4)
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/analytics/bodl-ga4.md:327]: ^URL to the KB doc will work once KB article is published

File: docs/api-docs/analytics/bodl-ga4.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+          });
+      }
+      if (typeof window.bodlEvents.checkout.orderPurchased === 'function') {
+          window.bodlEvents.checkout.orderPurchased(function(payload) {
+              gtag('event', GA_TO_BODL_CHECKOUT_EVENTS.purchase, transformPurchasePayload(payload));
+          });
+      }
+  }
+
+  if (window.addEventListener) {
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/analytics/bodl-ga4.md]: thanks, I fixed it

File: docs/api-docs/analytics/bodl-ga4.md (hunk 1 above)
Comment (solofeed) [reply docs/api-docs/analytics/bodl-ga4.md]: ping @jairo-bc

File: docs/api-docs/analytics/bodl-ga4.md (hunk 2 above)
Comment (bc-andreadao) [reply docs/api-docs/analytics/bodl-ga4.md]: I added a link to the embedded checkout tutorial (I'm not sure if there was anything specifically done to support embedded checkout)

File: docs/api-docs/analytics/bodl-ga4.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
+# Big Open Data Layer
+
+Big Open Data Layer (shortened as `BODL`, pronounced 'Bottle') is a global JavaScript object that allows BigCommerce to integrate with third-party analytics providers. This datalayer collects storefront data on shopper behavior and holds data needed for an analytic provider, such as GA4, to make analytic reports.  With BODL, providers can easily view and manage captured data from a standardized source. Providers can reformat the event data for their solutions.
+
+This guide demonstrates how you can integrate a BigCommerce store with [Google Analytics (GA4)](https://support.bigcommerce.com/s/article/GA4). Using BODL, you can track when a shopper starts checkout and purchases an order. You can track events for both redirected and [embedded checkout](/api-docs/storefronts/embedded-checkout/embedded-checkout-tutorial).
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/analytics/bodl-ga4.md:5]: ^URL to the KB doc will work once KB article is published

**Review by bc-traciporter — COMMENTED** [review]
Andrea, I suggested some editorial changes. Take the suggestions you feel comfortable with.

**Comment by bc-andreadao** [comment]
Closing this PR, as it has been replaced by [PR 1684](https://github.com/bigcommerce/dev-docs/pull/1684)

**Pull request closed without merging**